inst/stan/functions/discretised_logit_hazard.stan

```diff
@@ -42,54 +44,98 @@ vector adjust_lcdf_discrete(vector lcdf, int n, int max_strat) {
     // add to maximum value: cdf_n = 1
     adjusted_lcdf = lcdf;
     adjusted_lcdf[n] = 0;
+    // normalise to account for double censoring
+    adjusted_lcdf = adjusted_lcdf - log(2);
   } else if (max_strat == 2) {
-    // normalize: cdf = cdf / cdf[n]
-    adjusted_lcdf = lcdf - lcdf[n];
+    // normalize: cdf = cdf / (cdf[n] + cdf[n-1])
+    // both n and n-1 are in the denominator to account for 2 day width of the
+    // interval period
+    // If n were infinite this would be equivalent to dividing by 2.
+    if (n == 1) {
+      adjusted_lcdf = lcdf - lcdf[1];
+    }
+    if (n > 1) {
+      adjusted_lcdf = lcdf - log_sum_exp(lcdf[n], lcdf[n-1]);
+    }
   } else {
     reject("Unknown strategy to handle probability mass beyond the maximum value.");
   }
   return(adjusted_lcdf);
 }
 
-// compute discretised log scale pmf
-vector lcdf_to_log_prob(vector lcdf, int n) {
+// compute discretised log scale pmf from 0:n
+// assumes double censoring and a uniform interval approximation
+vector lcdf_to_uniform_double_censored_log_prob(vector lcdf, int n) {
   vector[n] lpmf; 
+  // p_0 = cdf_1 - cdf_0
   lpmf[1] = lcdf[1];
-  // p = cdf_n - cdf_n-1
-  lpmf[2:n] = log_diff_exp(lcdf[2:n], lcdf[1:(n-1)]);
+  if (n > 1) {
+    // p_1 = cdf_2 - cdf_0
+    lpmf[2] = lcdf[2];
+    if (n > 2) {
+      // p_n = cdf_n+1 - cdf_n-1
+      lpmf[3:n] = log_diff_exp(lcdf[3:n], lcdf[1:(n-2)]);
+    }
+  }
   return(lpmf);
 }
 
 // compute discretised logit hazard
-vector lcdf_to_logit_hazard(vector lcdf, int n) {
+vector lprob_to_uniform_double_censored_log_hazard(vector lprob, vector lcdf,
+   int n) {
   vector[n] lhaz;
-  vector[n-1] lccdf;
-  // cccdf = 1 - cdf
-  lccdf = log1m_exp(lcdf[1:(n-1)]);
+  // h_0 = cdf_1
   lhaz[1] = lcdf[1];
-  // h = 1 - CCDF_n / CCDF_n-1
-  lhaz[2:(n-1)] = log1m_exp(lccdf[2:(n-1)] - lccdf[1:(n-2)]);
+  // h_n = p_n / (1 - sum^{n-1}_{d=0} p_d)
+  // h_n = cdf_n+1 - cdf_n-1 / (1 - sum^{n-1}_{d=0} cdf_d+1 - cdf_d-1)
+  // h_n = cdf_n+1 - cdf_n-1 / 1 - sum^n_{d=0} cdf_d + sum^n-2_{d=1} cdf_d
```

Comment:
Hmm, I would have written `1 - sum^n_{d=1} cdf_d + sum^n-2_{d=0} cdf_d` here

Reply:
I think what is written is correct because the interval should be centred at the observed delay (for reasons discussed above). Given that context do you think what is written makes sense?

Reply:
Yes, I was only critical of this intermediate step in line 91. I changed the notation slightly and added brackets. The result is corrrect and makes sense.